internal/statuspage/statuspage.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package statuspage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strings"
+	"time"
+
+	frameworkDiag "github.com/hashicorp/terraform-plugin-framework/diag"
+	sdkv2Diag "github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+)
+
+const (
+	statuspageURL = "https://status.hashicorp.com/api/v1/summary"
+	warnSummary   = "You may experience issues using HCP."
+	warnDetailFmt = "HCP is reporting the following:\n\n%s\n\nPlease check https://status.hashicorp.com for more details."
+)
+
+// IDs from the incident.io status page for components relevant to this provider
+var hcpComponentNames = map[string]string{
+	"HCP API":           "01JK8R0BRY4185T4NHJFAXP35D",
+	"HCP Boundary":      "01JK8R0BRYHN4JYQ1H3WC42RWV",
+	"HCP Packer":        "01JK8R0BRYR9EYAGMNJ5EKC6CS",
+	"HCP Portal":        "01JK8R0BRYKPJS5K35R2ZCSHV0",
+	"HCP Vault Radar":   "01JK8R0BRYDYZFQH1V8ZSJKDFF",
+	"HCP Vault Secrets": "01JK8R0BRYY1ZM4NCA18A5T43A",
+	"HCP Waypoint":      "01JK8R0BRY0Q21819AYRKH5GZZ",
+}
+
+// These groups contain many regions and have a top-level ID that is not returned in the API response
+var hcpGroupNames = []string{
+	"HCP Consul Dedicated",
+	"HCP Vault Dedicated",
+}
+
+type statuspage struct {
```

Comment:
Can we add some comments with links to the definition of the status page structure?